[TEST-FAILURE] ClusterShutdownTest.cluster_mustBeShutDown_by_singleMember_when_clusterState_ACTIVE

```
java.lang.AssertionError: expected:<SHUT_DOWN> but was:<ACTIVE>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.hazelcast.cluster.ClusterShutdownTest$3.run(ClusterShutdownTest.java:97)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:828)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:842)
	at com.hazelcast.cluster.ClusterShutdownTest.assertNodesShutDownEventually(ClusterShutdownTest.java:93)
	at com.hazelcast.cluster.ClusterShutdownTest.testClusterShutdownWithSingleMember(ClusterShutdownTest.java:65)
	at com.hazelcast.cluster.ClusterShutdownTest.cluster_mustBeShutDown_by_singleMember_when_clusterState_ACTIVE(ClusterShutdownTest.java:26)
```

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-pr-builder/com.hazelcast$hazelcast/16376/testReport/junit/com.hazelcast.cluster/ClusterShutdownTest/cluster_mustBeShutDown_by_singleMember_when_clusterState_ACTIVE/

--- a/hazelcast/src/main/java/com/hazelcast/core/Cluster.java
+++ b/hazelcast/src/main/java/com/hazelcast/core/Cluster.java
@@ -182,6 +182,7 @@ public interface Cluster {
      * @see {@link com.hazelcast.instance.GroupProperty#CLUSTER_SHUTDOWN_TIMEOUT_SECONDS}
      *
      * @throws IllegalStateException    if member-list changes during the transaction
+     *                                  or there are ongoing/pending migration operations
      * @throws TransactionException     if there's already an ongoing transaction
      *                                  or this transaction fails
      *                                  or this transaction timeouts
@@ -206,6 +207,7 @@ public interface Cluster {
      *
      * @param transactionOptions transaction options
      * @throws IllegalStateException    if member-list changes during the transaction
+     *                                  or there are ongoing/pending migration operations
      * @throws TransactionException     if there's already an ongoing transaction
      *                                  or this transaction fails
      *                                  or this transaction timeouts

--- a/hazelcast/src/test/java/com/hazelcast/cluster/ClusterShutdownTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/cluster/ClusterShutdownTest.java
@@ -14,12 +14,12 @@ import org.junit.runner.RunWith;
 
 import java.util.concurrent.CountDownLatch;
 
-import static com.hazelcast.cluster.AdvancedClusterStateTest.changeClusterStateEventually;
 import static org.junit.Assert.assertEquals;
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelTest.class})
-public class ClusterShutdownTest extends HazelcastTestSupport {
+public class ClusterShutdownTest
+        extends HazelcastTestSupport {
 
     @Test
     public void cluster_mustBeShutDown_by_singleMember_when_clusterState_ACTIVE() {
@@ -49,38 +49,35 @@ public class ClusterShutdownTest extends HazelcastTestSupport {
     private void testClusterShutdownWithSingleMember(final ClusterState clusterState) {
         TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(4);
         final HazelcastInstance[] instances = factory.newInstances();
-        warmUpPartitions(instances);
 
-        changeClusterStateEventually(instances[0], clusterState);
+        final HazelcastInstance hz1 = instances[0];
 
-        final Thread thread = new Thread(new Runnable() {
-            @Override
-            public void run() {
-                instances[0].getCluster().shutdown();
-            }
-        });
+        hz1.getCluster().changeClusterState(clusterState);
+        hz1.getCluster().shutdown();
 
-        thread.start();
-
-        assertNodesShutDownEventually(instances);
+        for (final HazelcastInstance instance : instances) {
+            assertEquals(NodeState.SHUT_DOWN, getNode(instance).getState());
+        }
     }
 
     private void testClusterShutdownWithMultipleMembers(final int clusterSize, final int nodeCountToTriggerShutdown) {
         TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(clusterSize);
         final HazelcastInstance[] instances = factory.newInstances();
-        warmUpPartitions(instances);
 
-        changeClusterStateEventually(instances[0], ClusterState.PASSIVE);
+        instances[0].getCluster().changeClusterState(ClusterState.PASSIVE);
+
         final CountDownLatch latch = new CountDownLatch(1);
-        final Runnable shutdownRunnable = new Runnable() {
-            @Override
-            public void run() {
-                assertOpenEventually(latch);
-                instances[0].getCluster().shutdown();
-            }
-        };
 
         for (int i = 0; i < nodeCountToTriggerShutdown; i++) {
+            final HazelcastInstance instance = instances[i];
+            final Runnable shutdownRunnable = new Runnable() {
+                @Override
+                public void run() {
+                    assertOpenEventually(latch);
+                    instance.getCluster().shutdown();
+                }
+            };
+
             new Thread(shutdownRunnable).start();
         }
 

